README.md

```diff
@@ -41,26 +59,94 @@ struct MockPoolNode {
 }
 ```
 
+### Static Dispatching
+
+Nomos favours static dispatching over dynamic, influenced by Overwatch.
+This means you'll encounter Generics sprinkled throughout the codebase.
+While it might occasionally feel a bit over the top, it brings some solid advantages, such as:
+
+- Compile-time type checking
+- Highly modular and adaptable applications
+
+## Project Structure
+
+```
+nomos/
```

Comment:
nit: `nomos-node`

Reply:
Ah, the reason for that is @danielSanchezQ said we'll try to change the repo's name. Might have done the change too early, though. 
I'll wait for confirmation from him to know if I should roll it back to `nomos-node`.

Reply:
All good, no need to change if its planned.